src/applyConstructedStylesPatch.js

```diff
@@ -0,0 +1,52 @@
+// Helper to ensure our custom recorded array exists on the sheet.
+function ensureRecord(sheet) {
+  if (!sheet.__recordedCssRules) {
+    sheet.__recordedCssRules = [];
+  }
+}
+
+module.exports = function applyConstructedStylesPatch() {
+  if (typeof CSSStyleSheet === 'undefined') {
+    console.error('CSSStyleSheet is not supported in this browser');
+    return;
+  }
+  // Patch insertRule to record each rule string.
+  const originalInsertRule = CSSStyleSheet.prototype.insertRule;
+  CSSStyleSheet.prototype.insertRule = function (rule, index) {
+    ensureRecord(this);
+    // If index is not provided or invalid, default to appending.
+    if (index === undefined || index < 0 || index > this.__recordedCssRules.length) {
+      this.__recordedCssRules.push(rule);
+    } else {
+      this.__recordedCssRules.splice(index, 0, rule);
+    }
+    return originalInsertRule.call(this, rule, index);
+  };
+
+  // Patch deleteRule so that removed rules are taken out of our record.
+  const originalDeleteRule = CSSStyleSheet.prototype.deleteRule;
+  CSSStyleSheet.prototype.deleteRule = function (index) {
+    ensureRecord(this);
+    if (index >= 0 && index < this.__recordedCssRules.length) {
+      this.__recordedCssRules.splice(index, 1);
+    }
+    return originalDeleteRule.call(this, index);
+  };
+
+  // Patch replaceSync to capture the new CSS text.
+  const originalReplaceSync = CSSStyleSheet.prototype.replaceSync;
+  CSSStyleSheet.prototype.replaceSync = function (text) {
+    this.__recordedCssRules = text.split('\n').map((rule) => rule.trim());
```

Comment:
I noticed that the naming here assumes that each line is a separate rule. In the test that does seem to be the case, selectors aside, but I think in practice that won't always be true. It would be good to sort out the naming (line instead of rule perhaps) and to also add some multi-line styles to the tests.

Also, why split/map/trim here instead of something simpler like this?

```
this.__recordedCssRules = [text];
```

Oh, do we need deleteRule and insertRule with an index to work correctly after this is called? I bet that is not correct with this implementation. We might need a proper CSS parser for that to work.